ray-curator/ray_curator/examples/video/video_split_clip_example.py

```diff
@@ -0,0 +1,265 @@
+import argparse
+
+from ray_curator.backends.xenna import XennaExecutor
+from ray_curator.pipeline import Pipeline
+from ray_curator.stages.video.clipping.clip_extraction_stages import ClipTranscodingStage, FixedStrideExtractorStage
+from ray_curator.stages.video.io.video_download import VideoDownloadStage
+from ray_curator.stages.video.io.video_reader import VideoReaderStage
+
+
+def create_video_splitting_pipeline(args: argparse.Namespace) -> Pipeline:
+
+    # Define pipeline
+    pipeline = Pipeline(name="video_splitting", description="Split videos into clips")
+
+    # Add stages
+    pipeline.add_stage(VideoReaderStage(input_video_path=args.video_folder, video_limit=args.video_limit))
+    pipeline.add_stage(VideoDownloadStage(verbose=args.verbose))
+
+    if args.splitting_algorithm == "fixed_stride":
+        pipeline.add_stage(
+            FixedStrideExtractorStage(
+                clip_len_s=args.fixed_stride_split_duration,
+                clip_stride_s=args.fixed_stride_split_duration,
+                min_clip_length_s=args.fixed_stride_min_clip_length_s,
+                limit_clips=args.limit_clips,
+            )
+        )
+    else:
+        msg = f"Splitting algorithm {args.splitting_algorithm} not supported"
+        raise ValueError(msg)
+
+    pipeline.add_stage(ClipTranscodingStage(
+        num_cpus_per_worker=args.transcode_cpus_per_worker,
+        encoder=args.transcode_encoder,
+        encoder_threads=args.transcode_encoder_threads,
+        encode_batch_size=args.transcode_ffmpeg_batch_size,
+        use_hwaccel=args.transcode_use_hwaccel,
+        use_input_bit_rate=args.transcode_use_input_video_bit_rate,
+        num_clips_per_chunk=args.clip_re_chunk_size,
+        verbose=args.verbose,
+    ))
+
+    return pipeline
+
+
+def main(args: argparse.Namespace) -> None:
+
+    pipeline = create_video_splitting_pipeline(args)
+
+    # Print pipeline description
+    print(pipeline.describe())
+    print("\n" + "=" * 50 + "\n")
+
+    # Create executor
+    executor = XennaExecutor()
+
+    # Execute pipeline
+    print("Starting pipeline execution...")
+    pipeline.run(executor)
+
+    # Print results
+    print("\nPipeline completed!")
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    # General arguments
+    parser.add_argument("--video-folder", type=str, default="/home/aot/Videos")
+    parser.add_argument("--video-limit", type=int, default=-1, help="Limit the number of videos to read")
+    parser.add_argument("--verbose", action="store_true", default=False)
+    parser.add_argument("--output-clip-path", type=str, default="/mnt/mint/output")
+    parser.add_argument(
+        "--no-upload-clips",
```

Comment:
Is this argument used anywhere ?

Reply:
this is used when we have S3 client support, I just have it here in advance

Reply:
nit: i would say as general practice lets add things when the time comes, otherwise the diff becomes hard to reason about today (and in future). SImilarly in the `Video` dataclass you have more fields than what the "currently" merged stage need. This makes it hard to prune out what's needed where.

Reply:
+1 I think if we are not planning to support by the 25.09 release, then it should not be included yet.